### DISCUSSION TITLE
Why don't JSON types quite map to the databases type?

### DISCUSSION THREAD

**Discussion by jmg-duarte** [SeaQL/sea-orm#1004] [Q&A] [answered]
`ColumnDef::json` docs state:
> Set column type as json. On MySQL, this is equivalent to `json_binary`. On MariaDB, this is equivalent to `text`. On PgSQL, this is equivalent to `json`.

And `ColumnDef::json_binary` docs state:
> Set column type as json binary. On MySQL, this is equivalent to `json`. On MariaDB, this is equivalent to `text`. On PgSQL, this is equivalent to `jsonb`.

While for Postgres it makes sense, why are the types seemingly switched for MySQL?

**Comment by billy1624** [comment, marked as answer]
Hey @jmg-duarte, sorry for the confusion. If you look into the [source code](https://github.com/SeaQL/sea-query/blob/80f48951436a7e94c9886bd6d0ba0347af4c1ce5/src/backend/mysql/table.rs#L97-L98) of SeaQuery, both `ColumnDef::Json` and `ColumnDef::JsonBinary` map to `json` in MySQL.

I'll update the docs right away. And thank you for pointing this out :)
